### PR TITLE
Performance issues

### PR DESCRIPTION
**Pull request by mlopezFC**

### CODE REVIEW

**Comment by andersforsell** [comment]
Hi, just curious which performance issues are fixed in this PR?

**Comment by mlopezFC** [comment]
Hi @andersforsell , thanks for asking, I realized that I didn't create a ticket describing the issues before fixing them. These fixes are basically a backports of [this component](https://github.com/vaadin-component-factory/selection-grid-pro-flow). The issues briefly describing the problems are [this one](https://github.com/vaadin-component-factory/selection-grid-pro-flow/issues/8) and [this one](https://github.com/vaadin-component-factory/selection-grid-pro-flow/issues/11).

File: selection-grid-flow/src/main/resources/META-INF/resources/frontend/src/helpers.js [hunk 1]
```diff
@@ -35,12 +35,31 @@ export function _selectionGridSelectRow(e) {
         }
     }
 }
+export function _debounce(func, wait, immediate) {
+	debugger;
```
Comment (andersforsell) [inline selection-grid-flow/src/main/resources/META-INF/resources/frontend/src/helpers.js:39]: Were the debugger statements left intentionally?

Comment (mlopezFC) [reply selection-grid-flow/src/main/resources/META-INF/resources/frontend/src/helpers.js:39]: Good catch, probably not, will fix this right away. Thanks!

**Comment by andersforsell** [comment]
This PR breaks SelectionTreeGrid since `selectRangeOnlyOnClick(int fromIndex, int toIndex)` is not available there, see https://github.com/vaadin-component-factory/selection-grid-flow/issues/56

**Comment by randomuser1231232** [comment]
Hi @mlopezFC,
SelectionTreeGrid suffers from the same performance issues. Any plans to fix them there aswell?

**Pull request merged**